### PR TITLE
Terror spiders now have intro text

### PR DESCRIPTION
**Pull request by AzuleUtama**
## What Does This PR Do
Addresses the request in #19546 by adding in some introductory text for terror spiders that are either event spawned or taken control of by ghosts. The text aims to tell spiders their role, abilities, strengths and weaknesses rather than expecting them to either know it before taking the role or click the wiki link to read up.

<details>
<summary>The exact intro text for each spider is as follows:</summary>
  
### Red Terror
As a Red Terror Spider, your role is to directly attack the crew and defend weaker spiders. You have high health and deal high damage which increases the more injured you are. Your brute strength also allows you to force open powered doors and your webs are stronger and take longer for the crew to destroy. However, you move much slower than most other spiders.

### Green Terror
As a Green Terror Spider, your role is to lay and protect spider eggs so they can hatch and mature into more spiders. You can lay a new set of eggs for every 2 corpses you web, so work with other spiders to collect as many bodies as you can. To aid with this you have moderate health and deal moderate damage, with your bite and webs blurring the vision of any victims.

### Gray Terror
As a Gray Terror Spider, your role is to ambush prey by creating and waiting on webs. You create webs much faster than other spiders and said webs are also harder for the crew to see. While stood on your special web you are almost invisible and will deal double damage to prey unlucky enough to get webbed. However, you have low health and only deal moderate damage.

### Black Terror
As a Black Terror Spider, your role is to perform hit and run attacks with your venom. Anyone you bite or who falls into your webs will be injected with black terror venom, dealing heavy damage over time. Your webs will not inject victims wearing full hardsuits. However, you have low health and deal low direct damage, so you should avoid prolonged combat with the crew.

### White Terror
As a White Terror Spider, your role is to infect the crew with spider eggs which make them a host for future spiderlings. Both your bite and webs will inject these eggs into your victims, which will gradually cause spiderlings to burst out of them unless removed through surgery. Your webs will not inject victims wearing full hardsuits. You also have high health and while your attacks are weak, they will knock people down to make it harder for them to chase after you.

### Purple Terror
As a Purple Terror Spider, your role is to guard all princess or queen terror spiders. You move faster than other spiders, have high health, deal decent damage, can force open powered doors and can destroy walls. Additionally, your webs are thick and will block vision for most of the crew. However, being away from queen or princess spiders for too long will cause you to degenerate, taking gradual damage until you die or gain sight of them again.

### Brown Terror
As a Brown Terror Spider, your role is to breach areas for other spiders to attack. Your attacks are strong and in addition to being able to open powered doors you can also break down walls. You are also able to breach welded vents, allowing groups of terror spiders to break into areas that the crew might consider secure. However you have low health, move slower than other spiders and cannot spin any webs.

### Prince of Terror
As a Prince of Terror Spider, your role is to slaughter the crew with no remorse. You have extremely high health and deal more damage than any other terror spider, with your attacks also dealing high stamina damage and knocking down crew. You can also force open powered doors, smash down walls and lay webs which take longer to destroy while also blocking vision. However, your large size makes you unable to ventcrawl and you regenerate health slower than other spiders.

### Princess of Terror
As a Princess of Terror Spider, your role is to build and maintain a small nest. You constantly generate eggs at a slow rate which you can lay at a desired location, though you are limited mostly to basic terror spider eggs. Eggs generated while you are in a pipe will be lost. You can also open powered doors and your webs are airtight, being capable of blocking off exposure to space. You do not take damage from weaker weapons or projectiles. However, you only have moderate health and deal moderate damage, making you weak in direct fights and reliant on other spiders for defence.

### Mother of Terror
As a Mother of Terror Spider, your role is to support other spiders and spiderlings. You can pickup and incubate spiderlings to ensure they mature safely and at a faster rate, and your webs cannot be passed by spiderlings to generally keep them closer to nests. You can also produce royal jelly for other spiders to consume for faster health regeneration. Both of these require you to expend your own regeneration, though you gain much more from webbing corpses than other spiders. While you can open powered doors and bust open vents like a brown spider, you have low health and deal low damage, so should avoid fights wherever possible.

### Queen of Terror
As a Queen of Terror Spider, your role is build and maintain a large nest. You start off being able to ventcrawl like smaller spiders, but once you enter nest mode you will lose this in return for being able to constantly generate spider eggs to lay down as well as being able to smash down walls. Specialized egg types have a limit on how many can be laid, and you need to remain alive for some time before you can lay the strongest eggs. You have high health and are armored, with weaker attacks being unable to harm you. While your melee attacks only deal moderate damage, you additionally have access to a powerful ranged acid attack. You can also open powered doors and welded vents and your webs are airtight, being capable of blocking off exposure to space.

### Empress of Terror (Admin only)
As an Empress of Terror Spider, you role is to bring unholy terror to all living things. You more health than any other terror spider and deal extremely high damage to anything you attack. You can also lay eggs at an incredibly fast rate. You can also break through practically anything, so the crew will have zero hope of containing you. Have fun!

</details>

## Why It's Good For The Game
Giving players who are new to either terror spiders or a specific type of spider some basic information lets them know what they need to do and what their main strengths are.

## Images of changes
image

## Testing
Forced a T-spider event and confirmed the text shows up when you assume control of the spider.

## Changelog
:cl:
tweak: Terror spiders now have introductory text telling them their role and strengths.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/empress.dm [hunk 1]
```diff
@@ -11,6 +11,10 @@
 	name = "Empress of Terror"
 	desc = "The unholy offspring of spiders, nightmares, and lovecraft fiction."
 	spider_role_summary = "Adminbus spider"
+	spider_intro_text = "As an Empress of Terror Spider, you role is to bring unholy terror to all living things. \
+	You more health than any other terror spider and deal extremely high damage to anything you attack. \
```
Comment (farie82) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/empress.dm:15]:
```suggestion
	You have more health than any other terror spider and deal extremely high damage to anything you attack. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm [hunk 2]
```diff
@@ -40,6 +40,12 @@
 		to_chat(user, "<span class='notice'>Someone else already took this spider.</span>")
 		return
 	key = user.key
+	to_chat(src, "<b><font size=3 color='red'>You are a Terror Spider!</font></b>")
```
Comment (farie82) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm:43]: Dupe code my beloathed. It is better to make a `give_introduction` proc which does all the text here. Then call that in both cases

**Review by farie82 — CHANGES_REQUESTED** [review]

File: code/modules/events/spider_terror.dm [hunk 3]
```diff
@@ -68,6 +68,12 @@
 			S.forceMove(vent)
 			S.add_ventcrawl(vent)
 		SEND_SOUND(S, sound('sound/ambience/antag/terrorspider.ogg'))
+		to_chat(S, "<b><font size=3 color='red'>You are a Terror Spider!</font></b>")
+		to_chat(S, "Work with other terror spiders in your hive to eliminate the crew and claim the station as your nest!")
+		to_chat(S, "<b>Remember to follow the orders of higher tier spiders, such as the princess or queen.</b>")
```
Comment (S34NW) [inline code/modules/events/spider_terror.dm:73]:
```suggestion
		to_chat(S, "<b>Remember to follow the orders of higher tier spiders, such as princesses or queens.</b>")
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm [hunk 4]
```diff
@@ -12,8 +12,10 @@
 	name = "Black Terror spider"
 	desc = "An ominous-looking spider, black as the darkest night. It has merciless eyes, and a blood-red hourglass pattern on its back."
 	spider_role_summary = "Hit-and-run attacker with extremely venomous bite."
+	spider_intro_text = "As a Black Terror Spider, your role is to perform hit and run attacks with your venom. \
+	Anyone you bite or who falls into your webs will be injected with black terror venom, dealing heavy damage over time. \
+	However, you have low health and deal low damage, so you should avoid prolonged combat with the crew."
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm:17]:
```suggestion
	However, you have low health and deal low direct damage, so you should avoid prolonged combat with the crew."
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm [hunk 5]
```diff
@@ -11,6 +11,10 @@
 	name = "Brown Terror spider"
 	desc = "An ominous-looking spider, colored brown like the dirt it crawled out of. Its forearms have sharp digging claws."
 	spider_role_summary = "Vent-breaking spider that breaches into new areas."
+	spider_intro_text = "As a Brown Terror Spider, your role is to breach into areas for other spiders to attack. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm:14]:
```suggestion
	spider_intro_text = "As a Brown Terror Spider, your role is to breach areas for other spiders to attack. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm [hunk 6]
```diff
@@ -11,6 +11,10 @@
...
+	Your attacks are strong and in addition to being able to open powered doors you can also break down walls. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm:15]:
```suggestion
	Your attacks are strong, and in addition to being able to open powered doors you can also break down walls. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm [hunk 7]
```diff
@@ -11,6 +11,10 @@
...
+	Your unique skill is being able to breach welded vents, allowing groups of terror spiders to break into welded-off areas. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm:16]:
```suggestion
	Your are also able to breach welded vents, allowing groups of terror spiders to break into areas that the crew might consider secure. \
```
Not unique :(

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm [hunk 8]
```diff
@@ -11,6 +11,10 @@
...
+	However you have low health, move slower than other spiders and cannot spin any webs."
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/brown.dm:17]:
```suggestion
	However you have low health, move slower than other spiders, and cannot spin any webs."
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/empress.dm [hunk 9]
```diff
@@ -11,6 +11,10 @@
...
+	You can also lay eggs at an insanely fast rate, spreading even more chaos to everyone nearby. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/empress.dm:16]:
```suggestion
	You can lay eggs at an incredibly fast rate. \
```

Rest of this text is superfluous and not useful.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm [hunk 10]
```diff
@@ -40,6 +40,12 @@
...
+	to_chat(src, "Work with other terror spiders in your hive to eliminate the crew and claim the station as your nest!")
+	to_chat(src, "<b>Remember to follow the orders of higher tier spiders, such as the princess or queen.</b>")
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm:45]:
```suggestion
	to_chat(src, "<b>Remember to follow the orders of higher tier spiders, such as princesses or queens.</b>")
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/gray.dm [hunk 11]
```diff
@@ -12,6 +12,10 @@
 	name = "Gray Terror spider"
 	desc = "An ominous-looking gray spider. It seems to blend into webs, making it hard to see."
 	spider_role_summary = "Stealth spider that ambushes weak humans."
+	spider_intro_text = "As a Gray Terror Spider, your role is to ambush prey by creating and waiting inside webs. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/gray.dm:15]:
```suggestion
	spider_intro_text = "As a Gray Terror Spider, your role is to ambush prey by creating and waiting on webs. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/gray.dm [hunk 12]
```diff
@@ -12,6 +12,10 @@
...
+	You create webs much faster than other spiders and said webs are also harder for the crew to see. \
+	While laying in any web you are almost invisible and will deal double damage to prey unlucky enough to get webbed. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/gray.dm:17]:
```suggestion
	While stood on your special webs you are almost invisible and will deal double damage to prey. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/prince.dm [hunk 13]
```diff
@@ -12,6 +12,10 @@
 	name = "Prince of Terror spider"
 	desc = "An enormous, terrifying spider. It looks like it is judging everything it sees. Its hide seems armored, and it bears the scars of many battles."
 	spider_role_summary = "Miniboss terror spider. Lightning bruiser."
+	spider_intro_text = "As a Prince of Terror Spider, your role is to slaughter the crew with no remorse. \
+	You have extremely high health and your attacks deal massive brute damage, while also dealing high stamina damage and knocking down crew. \
+	You can also force open powered doors, smash down walls and lay webs which take longer to destroy while also blocking vision. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/prince.dm:17]:
```suggestion
	You can also force open powered doors, smash down walls and lay webs which take longer to destroy and block vision. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 14]
```diff
@@ -12,6 +12,10 @@
 	name = "Princess of Terror spider"
 	desc = "An enormous spider. It looks strangely cute and fluffy."
 	spider_role_summary = "Mini-Queen"
+	spider_intro_text = "As a Princess of Terror Spider, your role is to build and maintain a small nest. \
+	You constantly generate eggs at a slow rate which you can lay at a desired location, though you are limited mostly to basic terror spider eggs. \
+	You can also open powered doors and your webs are airtight, being capable of blocking off exposure to space. \
+	However, you only have moderate health and deal moderate damage, making you weak in direct fights and reliant on other spiders for defence."
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:18]:
```suggestion
	You do not take damage from weapons or projectiles that deal less than [force_threshold] damage. \
	However, you only have moderate health and deal moderate damage, making you weak in direct fights against gun users and reliant on other spiders for defence."
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/purple.dm [hunk 15]
```diff
@@ -12,6 +12,10 @@
 	name = "Purple Terror spider"
 	desc = "An ominous-looking purple spider. It looks about warily, as if waiting for something."
 	spider_role_summary = "Guards the nest of the Queen of Terror."
+	spider_intro_text = "As a Purple Terror Spider, your role is to guard all princess or queen terror spiders. \
+	You move faster than other spiders, have high health, deal decent damage, force open powered doors and can destroy walls. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/purple.dm:16]:
```suggestion
	You move faster than other spiders, have high health, deal decent damage, can force open powered doors and can destroy walls. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm [hunk 16]
```diff
@@ -12,6 +12,11 @@
 	name = "Queen of Terror spider"
 	desc = "An enormous, terrifying spider. Its egg sac is almost as big as its body, and teeming with spider eggs!"
 	spider_role_summary = "Commander of the spider forces. Lays eggs, directs the brood."
+	spider_intro_text = "As a Queen of Terror Spider, your role is build and maintain a large nest. \
+	You start off being able to ventcrawl like smaller spiders, but once you enter nest mode you will lose this in return for being able to constantly generate spider eggs to lay down as well as being able to smash down walls. \
+	Specialized egg types have a limit on how many can be laid, and you need to remain alive for some time before you can lay the strongest eggs. \
+	You have high health and are armored, with weaker attacks being unable to harm you. While your melee attacks only deal moderate damage, you additionally have access to a powerful ranged acid attack. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm:18]:
```suggestion
	You have high health and are armored, with weaker attacks (those under [force_threshold] damage) being unable to harm you. While your melee attacks only deal moderate damage, you additionally have access to a powerful ranged acid attack. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/red.dm [hunk 17]
```diff
@@ -13,6 +13,9 @@
 	name = "Red Terror spider"
 	desc = "An ominous-looking red spider, it has eight beady red eyes, and nasty, big, pointy fangs! It looks like it has a vicious streak a mile wide."
 	spider_role_summary = "High health, high damage, very slow, melee juggernaut"
+	spider_intro_text = "As a Red Terror Spider, your role is to directly attack the crew and defend weaker spiders. You have high health and deal high damage which increases the more injured you are. \
+	Your brute strength also allows you to force open powered doors and your webs are stronger and take longer for the crew to destroy. \
+	However, you move much slower than other spiders."
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/red.dm:18]:
```suggestion
	However, you move much slower than most other spiders."
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/white.dm [hunk 18]
```diff
@@ -11,6 +11,9 @@
 	name = "White Terror spider"
 	desc = "An ominous-looking white spider, its ghostly eyes and vicious-looking fangs are the stuff of nightmares."
 	spider_role_summary = "Rare, bite-and-run spider that infects hosts with spiderlings"
+	spider_intro_text = "As a White Terror Spider, your role is to infect the crew with spider eggs which make them a host for future spiderlings. \
+	Both your bite and webs will inject these eggs into your victims, which will gradually cause spiderlings to burst out of them unless removed through surgery. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/white.dm:15]:
```suggestion
	Both your bite and webs will inject these eggs into your victims, which will gradually cause spiderlings to burst out of them unless removed through surgery. Your webs will not inject victims wearing full hardsuits. \
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm [hunk 19]
```diff
@@ -12,8 +12,10 @@
 	name = "Black Terror spider"
 	desc = "An ominous-looking spider, black as the darkest night. It has merciless eyes, and a blood-red hourglass pattern on its back."
 	spider_role_summary = "Hit-and-run attacker with extremely venomous bite."
+	spider_intro_text = "As a Black Terror Spider, your role is to perform hit and run attacks with your venom. \
+	Anyone you bite or who falls into your webs will be injected with black terror venom, dealing heavy damage over time. \
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/black.dm:16]:
```suggestion
	Anyone you bite or who falls into your webs will be injected with black terror venom, dealing heavy damage over time. Your webs will not inject victims wearing full hardsuits.  \
```

**Review by S34NW — CHANGES_REQUESTED** [review]
Mostly good work, some additional changes to grammar and additional text/info to include please.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm (hunk 16 above)
Comment (S34NW) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm:18]: This won't actually work as it is not being done at runtime

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm (hunk 14 above)
Comment (S34NW) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:18]: This won't actually work as it is not being done at runtime

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm [hunk 20]
```diff
@@ -12,6 +12,11 @@
 	name = "Mother of Terror spider"
 	desc = "An enormous spider. Tiny spiderlings are crawling all over it. Their beady little eyes all stare at you. The horror!"
 	spider_role_summary = "Carries spiderlings to protect them. Creates royal jelly that heals other spiders."
+	spider_intro_text = "As a Mother of Terror Spider, your role is to support other spiders and spiderlings. \
+	You can pickup and incubate spiderlings to ensure they mature safely and at a faster rate, and your webs cannot be passed by spiderlings to keep them closer to nests. \
+	You can also produce royal jelly for other spiders to consume for faster health regeneration. \
+	Both of these require you to expend your own regeneration, though you gain much more from webbing corpses than other spiders. \
+	While you can open powered doors and bust open vents like a brown spider, you have low health and deal low damage, so should avoid fights wherever possible."
```
Comment (SteelSlayer) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm:19]:
```suggestion
	While you can open powered doors and bust open vents like a brown spider, you have low health and deal low damage, so you should avoid fights wherever possible."
```

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm [hunk 21]
```diff
@@ -339,6 +340,13 @@ GLOBAL_LIST_EMPTY(ts_spiderling_list)
 		GLOB.ts_death_last = world.time
 		GLOB.ts_count_alive_station--
 
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/give_intro_text()
+	to_chat(src, "<center><span class='userdanger'>You are a Terror Spider!</span></center>")
+	to_chat(src, "<center>Work with other terror spiders in your hive to eliminate the crew and claim the station as your nest!")
```
Comment (SteelSlayer) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm:345]: Needs spans

Comment (S34NW) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm:345]: Even with no desired class?

Comment (SteelSlayer) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm:345]: Ah yeah, we default to the `notice` class in that case.

Also just noticed that this line needs a closing `</center>`.

**Review by SteelSlayer — APPROVED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**